Mono.Cecil/AssemblyReader.cs

```diff
@@ -243,12 +243,20 @@ void ReadGenericParameters (IGenericParameterProvider provider)
 				var parameter = parameters [i];
 
 				if (parameter.HasConstraints)
-					Mixin.Read (parameter.Constraints);
+					ReadGenericParameterConstraints (parameter);
 
 				ReadCustomAttributes (parameter);
 			}
 		}
 
+		void ReadGenericParameterConstraints (GenericParameter parameter)
+		{
+			var constraints = parameter.Constraints;
+
+			for (int i = 0; i < constraints.Count; i++)
```

Comment:
@jbevain out of curiosity, why do u prefer for over foreach?

Reply:
This codebase was started at a moment where `mcs` didn't generate great code for `foreach` on generic collections, and so used `for` instead of `foreach`. It's now used only for consistency with the rest of the code.